src/com/google/javascript/jscomp/Es6ConvertSuper.java

```diff
@@ -158,42 +173,50 @@ public boolean apply(Node n) {
       // Calls to super() constructors will be transpiled by Es6ConvertSuperConstructorCalls
       // later.
       return;
-    }
-    if (enclosingMemberDef.isStaticMember()) {
-      Node callTarget;
-      potentialCallee.detach();
-      if (potentialCallee == node) {
-        // of the form super()
-        // TODO(bradfordcsmith): This should report an error since this is not allowed by the
-        //     current spec.
-        potentialCallee =
-            IR.getprop(superName.cloneTree(), IR.string(enclosingMemberDef.getString()));
-        enclosingCall.putBooleanProp(Node.FREE_CALL, false);
-      } else {
-        // of the form super.method()
-        potentialCallee.replaceChild(node, superName.cloneTree());
+    } else if (enclosingMemberDef.isStaticMember()) {
+      if (callTarget.isSuper()) {
+        compiler.report(JSError.make(node, INVALID_SUPER));
```

Comment:
Before doing this PR, I think we'd better have one that removes support for calling `super()` in a non-constructor method.